yjit/src/codegen.rs

```diff
@@ -3588,7 +3623,7 @@ fn jit_rb_str_bytesize(
     _cme: *const rb_callable_method_entry_t,
     _block: Option<IseqPtr>,
     _argc: i32,
-    _known_recv_class: *const VALUE,
+    _known_recv_object: VALUE,
```

Comment:
Can you also rename all these to `comptime_recv` for consistency?

Reply:
Hmmmm wait. I think I'm just realizing, we might be doing this wrong.

`known_recv_klass` means we know the receiver class *for sure*. It means `jit_guard_known_klass` has been run.

`comptime_recv` is the receiver we see at compilation time. There is no guarantee of anything there.

So it doesn't make sense to replace `known_recv_klass` by the compile time receiver. I think we need both fields for this to be correct.